test/plugin/base_test.rb

```diff
@@ -859,46 +859,57 @@ def test_dataproc_log
     verify_log_entries(1, DATAPROC_PARAMS, 'jsonPayload')
   end
 
-  def test_http_request_from_record
-    setup_gce_metadata_stubs
-    setup_logging_stubs do
-      d = create_driver
-      d.emit('httpRequest' => HTTP_REQUEST_MESSAGE)
-      d.run
-    end
-    verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
-      assert_equal HTTP_REQUEST_MESSAGE, entry['httpRequest'], entry
-      assert_nil get_fields(entry['jsonPayload'])['httpRequest'], entry
+  def test_subfields_from_record
```

Comment:
Can we keep these fields in separate unit tests so that when tests fail, it's more obvious what we are testing? Something like:

```
  def test_log_entry_location_field_from_record
    verify_subfields_from_record(DEFAULT_HTTP_REQUEST_KEY)
  end

  def test_log_entry_location_field_from_record
    verify_subfields_from_record(DEFAULT_SOURCE_LOCATION_KEY)
  end

  def test_log_entry_location_field_from_record
    verify_subfields_from_record(DEFAULT_OPERATION_KEY)
  end

  // Below can be in the private section.
  def verify_subfields_from_record(payload_key)
    destination_key, payload_value = LOG_ENTRY_SUBFIELDS_PARAMS[payload_key]
    @logs_sent = []
    setup_gce_metadata_stubs
    setup_logging_stubs do
      d = create_driver
      d.emit(payload_key => payload_value)
      d.run
    end
    verify_log_entries(1, COMPUTE_PARAMS, destination_key) do |entry|
      assert_equal payload_value, entry[destination_key], entry
      fields = get_fields(entry['jsonPayload'])
      assert_nil fields[payload_key], entry
    end
  end

...

  LOG_ENTRY_SUBFIELDS_PARAMS = {
    # payload key, destination key, payload value
    DEFAULT_HTTP_REQUEST_KEY => ['httpRequest', HTTP_REQUEST_MESSAGE],
    DEFAULT_SOURCE_LOCATION_KEY => ['sourceLocation', SOURCE_LOCATION_MESSAGE],
    DEFAULT_OPERATION_KEY => ['operation', OPERATION_MESSAGE]
  }
``